components/tabs/src/sync/store.rs

```diff
@@ -92,16 +117,37 @@ impl<'a> Store for TabsStore<'a> {
                     continue;
                 }
             };
-            // TODO: this is wrong anything that doesn't use the sync manager crate,
-            // we need to get fxa_client_id from the clients collection instead.
             let id = record.id.clone();
-            remote_tabs.push(ClientRemoteTabs::from_record(id, record));
+            let tab = if let Some(remote_client) = self.remote_clients.borrow().get(&id) {
+                ClientRemoteTabs::from_record_with_device(id, remote_client, record)
+            } else {
+                ClientRemoteTabs::from_record(id, record)
+            };
+            remote_tabs.push(tab);
         }
         self.storage.replace_remote_tabs(remote_tabs);
         let mut outgoing = OutgoingChangeset::new("tabs".into(), inbound.timestamp);
         if let Some(local_tabs) = self.storage.get_local_tabs() {
+            let (client_name, device_type) = self
+                .remote_clients
+                .borrow()
+                .get(local_id)
+                .map(|client| {
+                    (
+                        client.device_name.clone(),
+                        client.device_type.unwrap_or(DeviceType::Mobile),
+                    )
+                })
+                .unwrap_or_else(|| {
+                    // TODO: Since we know our own client ID thanks to
+                    // `self.remote_clients`, we don't need consumers to
+                    // pass it themselves.
+                    (String::new(), DeviceType::Mobile)
```

Comment:
Since we have access to the clients engine, in theory, we don't need to pass `local_id` to the `TabsStore` at all—we already know which client record is ours. However, that would make it impossible to upload our own tabs when using this component without the sync manager.